Userland/Games/Chess/ChessWidget.cpp

```diff
@@ -293,8 +294,22 @@ void ChessWidget::mouseup_event(GUI::MouseEvent& event)
             move.promote_to = promotion_dialog->selected_piece();
     }
 
+    if (board().moves().size() == 0) {
+        if (!m_timer->is_active() && !m_unlimited_time_control) {
+            m_timer->start();
+        }
+    }
+
     if (board().apply_move(move)) {
         update_move_display_widget(m_board);
+        if (!m_unlimited_time_control) {
+            if (move.piece.color == Chess::Color::White) {
+                m_white_time_elapsed -= m_time_control_increment;
+            } else {
+                m_black_time_elapsed -= m_time_control_increment;
+            }
+            update_time_labels(m_time_control_seconds - m_white_time_elapsed, m_time_control_seconds - m_black_time_elapsed);
+        }
```

Comment:
Not sure if we should care or if I even read that correctly, but I think that the engine won't get increments.

Reply:
That was an oversight on my part. Do you think I should move it out to its own method (e.g. `apply_increment_to_move`) or just copy that over to the engine move code?

Reply:
That was an oversight on my part, now the engine can lose on time 😄